src/MemoryTracing/MemoryTracingUtils.cpp

```diff
@@ -85,4 +88,137 @@ std::optional<SystemMemoryUsage> GetSystemMemoryUsage() noexcept {
+      // According to the kernel code
+      // https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/fs/proc/task_mmu.c,
+      // the size unit of the VmRSS field in the /proc/<PID>/status file is fixed to "kB", which
+      // implies 1024 Bytes. We report in "KB" in ProcessSystemUsage.
+      CHECK(splits.size() == 3 && splits[2] == "kB");
+      int64_t rss_kb;
+      if (absl::SimpleAtoi(splits[1], &rss_kb)) return rss_kb;
+      return kMissingInfo;
+    }
+  }
+
+  return kMissingInfo;
+}
+
+std::optional<ProcessMemoryUsage> GetProcessMemoryUsage(uint32_t pid) noexcept {
+  uint64_t current_timestamp_ns = orbit_base::CaptureTimestampNs();
+
+  // Extract the VmRSS value in kilobytes from the /proc/<pid>/status file.
+  ErrorMessageOr<std::string> reading_result =
+      orbit_base::ReadFileToString(absl::StrFormat("/proc/%d/status", pid));
+  if (!reading_result.has_value()) {
+    ERROR("%s", reading_result.error().message());
+    return std::nullopt;
+  }
+  int64_t vm_rss_kb = GetVmRssFromProcessStatus(reading_result.value());
+
+  ProcessMemoryUsage process_memory_usage;
+  process_memory_usage.set_pid(pid);
+  process_memory_usage.set_timestamp_ns(current_timestamp_ns);
+  process_memory_usage.set_vm_rss_kb(vm_rss_kb);
+
+  return process_memory_usage;
+}
+
+std::string GetProcessMemoryCGroupName(std::string_view cgroup_content) {
+  if (cgroup_content.empty()) return "";
+
+  std::vector<std::string> lines = absl::StrSplit(cgroup_content, '\n');
+  if (lines.empty()) return "";
+
+  for (std::string_view line : lines) {
+    std::vector<std::string> splits = absl::StrSplit(line, absl::MaxSplits(':', 2));
+    CHECK(splits.size() == 3);
+    // Return the cgroup name without the leading "/".
+    if (splits[1] == "memory") return splits[2].substr(1);
+  }
+
+  return "";
+}
+
+int64_t GetCGroupMemoryLimitInBytes(std::string_view memory_limit_in_bytes_content) {
+  if (memory_limit_in_bytes_content.empty()) return kMissingInfo;
+
+  // The memory.limit_in_bytes file use "bytes" as the size unit.
+  int64_t memory_limit_in_bytes;
+  if (!absl::SimpleAtoi(memory_limit_in_bytes_content, &memory_limit_in_bytes)) return kMissingInfo;
+
+  return memory_limit_in_bytes;
+}
+
+int64_t GetRssFromCGroupMemoryStat(std::string_view memory_stat_content) {
+  if (memory_stat_content.empty()) return kMissingInfo;
+
+  std::vector<std::string> lines = absl::StrSplit(memory_stat_content, '\n');
+  if (lines.empty()) return kMissingInfo;
+
+  for (std::string_view line : lines) {
+    std::vector<std::string> splits = absl::StrSplit(line, ' ', absl::SkipWhitespace{});
+    CHECK(splits.size() == 2);
+    if (splits[0] == "rss") {
+      // According to the document https://www.kernel.org/doc/Documentation/cgroup-v1/memory.txt,
+      // the memory.stat file use "bytes" as the size unit.
+      int64_t rss_in_bytes;
+      if (!absl::SimpleAtoi(splits[1], &rss_in_bytes)) return kMissingInfo;
```

Comment:
Similarly here, this should be consistent with other instances of this pattern.

Reply:
Changed accordingly. Thanks!